src/ThriftSQL/Utils/Iterator.php

```diff
@@ -20,34 +22,49 @@ class Iterator implements \Iterator {
   private $buffer;
   private $location;
 
-  private $runCount = 0;
+  private $queryCount = 0;
+  private $rewindCount = 0;
   private $allowRerun = false;
+  private $useSchema = false;
 
   public function __construct( \ThriftSQL $thriftSQL, $queryStr ) {
     $this->thriftSQL = $thriftSQL;
     $this->queryStr = $queryStr;
+
+    $this->queryCount = 1;
+    $this->doQuery();
```

Comment:
I don't think we should run the query on `__construct()` let's keep the constructor clean and keep the delayed execution capability of the iterator.